x/nexus/keeper/general_message.go

```diff
@@ -125,3 +174,22 @@ func (k Keeper) getMessages(ctx sdk.Context) (generalMessages []exported.General
 
 	return generalMessages
 }
+
+// GetApprovedMessages returns up to limit approved messages where chain is the destination chain
+func (k Keeper) GetApprovedMessages(ctx sdk.Context, chain exported.ChainName, limit int64) []exported.GeneralMessage {
+	msgs := []exported.GeneralMessage{}
+	ids := []exported.MessageID{}
+	iter := k.getStore(ctx).IteratorNew(approvedGeneralMessagePrefix.Append(key.From(chain)))
+	defer utils.CloseLogError(iter, k.Logger(ctx))
```

Comment:
See the Paginated query for transfers to follow the same pattern for a limited read. Also, newlines.